File: C-structless/game_runner.c
#include <stdlib.h>
#include <time.h>
#include "game.h"


int
main ()
{
  newGame();

  add ("Chet");
  add ( "Pat");
  add ( "Sue");

	srand ((unsigned)time(0));

      do
	{
	  roll ( rand () % 5 + 1);

	  if (rand () % 9 == 7)
	    {
	      wrong_answer ();
	    }
	  else
	    {
	      was_correctly_answered ();
	    }
	}
      while (not_a_winner);
  }

File: kotlin/src/main/kotlin/com/adaptionsoft/games/trivia/runner/GameRunner.kt
package com.adaptionsoft.games.trivia.runner

import com.adaptionsoft.games.uglytrivia.Game
import java.util.*

object GameRunner {
    var notAWinner: Boolean = false
}

fun main(args: Array<String>) {
    val aGame = Game()

    aGame.add("Chet")
    aGame.add("Pat")
    aGame.add("Sue")

    val rand = Random()

    do {

        aGame.roll(rand.nextInt(5) + 1)

        if (rand.nextInt(9) == 7) {
            GameRunner.notAWinner = aGame.wrongAnswer()
        } else {
            GameRunner.notAWinner = aGame.wasCorrectlyAnswered()
        }


    } while (GameRunner.notAWinner)

}
